presto-native-execution/presto_cpp/main/connectors/tpcds/DSDGenIterator.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+
+#include "presto_cpp/external/dsdgen/include/dsdgen-c/dist.h"
+#include "presto_cpp/external/dsdgen/include/tpcds_constants.hpp"
+#include "presto_cpp/main/connectors/tpcds/utils/append_info-c.hpp"
+
+using namespace facebook::velox;
+namespace facebook::presto::connector::tpcds {
+
+typedef int64_t ds_key_t;
+
+typedef int (*tpcds_builder_func)(void*, ds_key_t, DSDGenContext& dsdgenCtx);
+
+void InitializeDSDgen(
+    double scale,
+    vector_size_t parallel,
+    vector_size_t child,
+    DSDGenContext& dsdGenContext);
+
+std::string getQuery(int query);
+
+/// This class exposes a thread-safe and reproducible iterator over TPC-DS
+/// synthetically generated data, backed by DSDGEN.
+class DSDGenIterator {
+ public:
+  explicit DSDGenIterator(
+      double scaleFactor,
+      vector_size_t parallel,
+      vector_size_t child);
+
+  void initializeTable(const std::vector<VectorPtr>& children, int table);
```

Comment:
Please add comments for each of the methods in this class as this is the final interface seen over dsdgen